lib/controllers/items.js

```diff
@@ -1,5 +1,46 @@
 const { Router } = require('express');
+const authenticate = require('../middleware/authenticate');
+const authorize = require('../middleware/authorize');
+const Item = require('../models/Item');
 
-module.exports = Router();
+module.exports = Router()
+  .get('/', authenticate, async (req, res, next) => {
+    try {
+      const list = await Item.getAll(req.user.id);
+      res.json(list);
+    } catch (e) {
+      next(e);
+    }
+  })
+
+  .put('/:id', [authenticate, authorize], async (req, res, next) => {
+    try {
+      const newItem = await Item.updateById(req.params.id, req.body);
+      if (!newItem) next();
+      res.json(newItem);
+    } catch (e) {
+      next(e);
+    }
+  })
+
+  .post('/', authenticate, async (req, res, next) => {
+    try {
+      const newItem = await Item.insert({ ...req.body, user_id: req.user.id });
+      res.json(newItem);
+    } catch (e) {
+      next(e);
+    }
+  })
+
+  .delete('/:id', async (req, res, next) => {
```

Comment:
In the same way do do for your put route, you should be running your delete route through the `[authenticate, authorize]` middleware.